hpi-swt2/lecture-portal
# Feature/#158 upload files to lectures

# Resolves #158 and #154.

This PR introduces the feature of uploading files to lectures and courses.

Acceptance Criteria:
From 154:
- [x] In the course overview, there is a button titled "Add new material" near the box that displays student and lecturer material associated with that course
- [x] Clicking the button will open a window where the user can choose a file to upload to the course and then upload it by clicking another button
- [x] When the file has been uploaded, all information needed is automatically entered into the DB (timestamp, author, ...)
From 158:
- [x] In the lecture settings view - and that ALSO means in the lecture creation view (since they are the same) - is a section "materials" 
- [x] In this section is a list of all materials associated to this particular lecture
- [x] Clicking on a file's name will automatically open (download / open in browser if link) the file
- [x] There is a button / text "delete file" besides every file, that when clicked promts a window titled "Are you sure you want to remove this file?", a "Yes" and a "No" option.
- [x] Clicking "Yes" removes the file entirely from the lecture, course and the DB, and closes the window
- [x] Clicking "No" Closes the window as well
- [x] There is a button "Add lecture material" in the section through wich the lecturer can add and upload new material to the lecture
- [x] Adding /uploading anew file will automatically fill out all information about the file in the db
- [x] A student will NEVER be able to upload, nor to delete materials that are associated to a specific LECTURE

Testing steps:
- go to course page as student or lecturer
- click "Add new material"
- upload something, verify you can see, download, delete it
- go to lecture page
- click "Add lecture material"
- upload something, verify you can see, download, delete it

Steps needed for migration:
- None

---

Definition Of Done:
